cli/cmd/plugin/ui/web/tanzu-ui/src/views/unmanaged-cluster/UnmanagedClusterInventory.test.tsx

```diff
@@ -1,17 +1,142 @@
 // React imports
 import React from 'react';
-import { BrowserRouter } from 'react-router-dom';
 
 // Library imports
-import { render } from '@testing-library/react';
-import '@testing-library/jest-dom';
+import { fireEvent, render, screen, waitFor } from '@testing-library/react';
+import { rest } from 'msw';
+import { setupServer } from 'msw/lib/node';
 
 // App imports
 import UnmanagedClusterInventory from './UnmanagedClusterInventory';
 
-describe('UnmanagedClusterLanding', () => {
-    test('should render', () => {
-        const view = render(<UnmanagedClusterInventory />, { wrapper: BrowserRouter });
-        expect(view).toBeDefined();
+jest.mock('react-router-dom', () => ({
+    ...(jest.requireActual('react-router-dom') as any),
+    useNavigate: () => jest.fn(),
+}));
+
+describe('UnmanagedClusterInventory component', () => {
+    const server = setupServer(
+        rest.get('/api/unmanaged', (req, res, ctx) => {
+            return res(
+                ctx.status(200),
+                ctx.json([
+                    {
```

Comment:
I would suggest setting up a const which is an array of these objects, then you can use the array data in your tests below